### PR TITLE
Enhance cilium fetcher with rules

### PR DESCRIPTION
**Pull request by mrproliu**
Due to Cilium monitoring all services in the same cluster, and it will result in a large amount of unnecessary data. Therefore, the following parts have been introduced:
1. Exclusion policy: When meeting the exclusion policy criteria, some traffic will be ignored to reduce data generation.
2. Customize metadata: In previous designs, automatic use of ALS metadata mapping files was used; here we reuse a new metadata mapping file instead with unchanged rules.
3. Support for Hierarchy function provisioning: Allowing it to point towards Kubernetes monitoring in hierarchies

Since Cilium Fetcher is a new feature in this milestone and has already been added to the changelog, there's no need for re-addition.

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [ ] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

File: oap-server/server-starter/src/main/resources/hierarchy-definition.yml [hunk 1]
```diff
@@ -112,3 +115,5 @@ layer-levels:
 
   K8S_SERVICE: 0
 
+  CILIUM_SERVICE: 1
+
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/hierarchy-definition.yml:119]: Please order the services based on priority.

File: oap-server/server-starter/src/main/resources/hierarchy-definition.yml [hunk 2]
```diff
@@ -74,6 +74,9 @@ hierarchy:
     KAFKA: lower-short-name-with-fqdn
     PULSAR: lower-short-name-with-fqdn
 
+  CILIUM_SERVICE:
+    K8S_SERVICE: short-name
+
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/hierarchy-definition.yml:79]: Please update doc accordingly, https://skywalking.apache.org/docs/main/next/en/concepts-and-designs/service-hierarchy-configuration/

File: oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+namespaces:
+    - kube-system
+
+labels:
+    - k8s:io.cilium.k8s.namespace.labels.istio-injection: enabled
+      k8s:security.istio.io/tlsMode: istio
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml:22]: Does this rule mean `excluding istio relative services`? If so, I should consider whether this should be in default configurations.

File: oap-server/server-starter/src/main/resources/hierarchy-definition.yml (hunk 2 above)
Comment (mrproliu) [reply oap-server/server-starter/src/main/resources/hierarchy-definition.yml:79]: updated.

File: oap-server/server-starter/src/main/resources/hierarchy-definition.yml (hunk 1 above)
Comment (mrproliu) [reply oap-server/server-starter/src/main/resources/hierarchy-definition.yml]: updated.

File: oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml (hunk 3 above)
Comment (mrproliu) [reply oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml:22]: Yes, this configuration means excluding the services that have been added the proxy sidecar.

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml:22]: Please add comments, and we should document this in the docs about cilium monitoring would be disabled automatically if services have been managed by mesh.

Comment (mrproliu) [reply oap-server/server-starter/src/main/resources/cilium-rules/exclude.yaml:22]: updated, please check.

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**